src/expr/dtype.h

```diff
@@ -489,6 +499,47 @@ class DType
    * Helper for mkGroundTerm and mkGroundValue above.
    */
   Node mkGroundTermInternal(TypeNode t, bool isValue) const;
+  /**
+   * This method is used to get strict subfield types of this datatype.
+   *
+   * A subfield type T of a datatype type D is a type such that a value of
+   * type T may appear as a subterm of a value of type D.
+   *
+   * A *strict* subfield type T of a datatype type D is a type such that a
+   * value v of type T may appear as a subterm of a value of D, and moreover
+   * v occurs as a strict subterm of a non-datatype term in that value.
+   *
+   * For example, the strict subfield types of:
+   *   T -> Emp | Container(s : (Set List))
+   *   List -> nil | cons( head : Int, tail: List)
+   * are { List, Int }. Notice that Int is a strict subfield type since it
+   * appears as a subfield type of List, and List is a strict subfield type
+   * of T. In other words, Int is a strict subfield type due to the above
+   * definition due to the term (Container (singleton (cons 0 nil))), where
+   * 0 occurs as a subterm of (singleton (cons 0 nil)).
+   *
+   * For example, the strict subfield types of:
+   *   T -> Emp | Container(s : List)
+   *   List -> nil | cons( head : (Set T), tail: List)
+   * are { T, List }. Notice that T is a strict subfield type of itself since
+   * List is a subfield type of T and T is a strict subfield type of List.
+   *
+   * For example, the strict subfield types of:
```

Comment:
Thanks for all these examples. Could you also specify the non-strict subfield types in the examples?

Reply:
Sure.